### DISCUSSION TITLE
miner start failed

### DISCUSSION THREAD

**Discussion by lbj2004032** [filecoin-project/lotus#10496] [Storage Provider]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus fvm/fevm - Lotus FVM and FEVM interactions
- [ ] lotus miner/worker - sealing
- [X] lotus miner - proving(WindowPoSt/WinningPoSt)
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [X] Other

### Lotus Version

```text
lotus version 1.21.0-dev+mainnet+git.64b9b532c
lotus-miner version 1.21.0-dev+mainnet+git.64b9b532c
```

### Repro Steps

lotus-miner run

### Describe the Bug

start miner block at 
2023-03-16T22:24:35.165+0800    INFO    stores  paths/index.go:181      New sector storage: 12db6be1-710e-4dbb-9be6-3618dcd916cd

### Logging Information

```text
f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params is ok
2023-03-16T22:24:35.136+0800    INFO    paramfetch      go-paramfetch@v0.0.4/paramfetch.go:209  Parameter file /mnt/lotus1/data_v28/filecoin-proof-parameters/v28-proof-of-spacetime-fallback-merkletree-poseidon_hasher-8-8-0-0377ded656c6f524f1618760bffe4e0a1c51d5a70c4509eedae8a27555733edc.params is ok
2023-03-16T22:24:35.137+0800    INFO    paramfetch      go-paramfetch@v0.0.4/paramfetch.go:233  parameter and key-fetching complete
2023-03-16T22:24:35.165+0800    INFO    stores  paths/index.go:181      New sector storage: 12db6be1-710e-4dbb-9be6-3618dcd916cd
```

**Comment by Reiers** [comment]
Hi @lbj2004032 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Reply by lbj2004032** [reply]
> Hi @lbj2004032
> 
> Thanks for the report. The issue you are experiencing lies elsewhere.
> 
> Moving this issue to Lotus Discussions for help and troubleshooting.
> 
> Closing Lotus issue ticket.

This is the calculation power of the other three nodes, which is tens of P

Is there any way to start it

**Reply by lbj2004032** [reply]
> Hi @lbj2004032
> 
> Thanks for the report. The issue you are experiencing lies elsewhere.
> 
> Moving this issue to Lotus Discussions for help and troubleshooting.
> 
> Closing Lotus issue ticket.

![1679008218212](https://user-images.githubusercontent.com/101935091/225772431-43d7333b-1e93-4448-b51f-0e76c277e896.png)
There has been a long time of continuous data download, but this step cannot be skipped. Is there a possibility of a dead cycle?

**Comment by Reiers** [comment]
Looks like you are running on `master`, could you try to revert back to the latest stable release?
Like this:
```
1. cd lotus/
2. git pull
3. git checkout v1.20.3
4. git submodule update
5. make clean
6. RUSTFLAGS="-C target-cpu=native -g" FFI_BUILD_FROM_SOURCE=1 make all
7. sudo make install
```
Thank you !

**Reply by lbj2004032** [reply]
> Looks like you are running on `master`, could you try to revert back to the latest stable release? Like this:
> 
> ```
> 1. cd lotus/
> 2. git pull
> 3. git checkout v1.20.3
> 4. git submodule update
> 5. make clean
> 6. RUSTFLAGS="-C target-cpu=native -g" FFI_BUILD_FROM_SOURCE=1 make all
> 7. sudo make install
> ```
> 
> Thank you !
I found the stuck line of code but not the reason
It is guessed that there should be an infinite loop. By printing the output file name, it is found that there is an infinite loop.
Problem code:
storage/paths/local.go, where the function: declareSectors
When reading the permanently stored cache directory, the code gets stuck at the line os.ReadDir.

dir_unix.go
![image](https://user-images.githubusercontent.com/101935091/228479356-981620bf-950e-4bd8-afab-30c5d9d814c6.png)

local.go
![image](https://user-images.githubusercontent.com/101935091/228478873-27ef7e1d-5941-4793-a37c-d4cc76208e0d.png)
